docs/api-diffs/9.0/rc.1/diff.md

```diff
@@ -0,0 +1,1290 @@
++            public ExecuteCommandContext();
++            public required CancellationToken CancellationToken { get; set; }
++            public required string ResourceName { get; set; }
++            public required IServiceProvider ServiceProvider { get; set; }
++        }
++        public class ExecuteCommandResult {
++            public ExecuteCommandResult();
++            public string ErrorMessage { get; set; }
++            public required bool Success { get; set; }
++        }
++        [DebuggerDisplayAttribute("Type = {GetType().Name,nq}, Key = {Key}")]
++        public class HealthCheckAnnotation : IResourceAnnotation {
++            public HealthCheckAnnotation(string key);
++            public string Key { get; }
++        }
++        public sealed class HealthReportSnapshot : IEquatable<HealthReportSnapshot> {
++            public HealthReportSnapshot(string Name, HealthStatus Status, string Description, string ExceptionText);
++            public string Description { get; set; }
++            public string ExceptionText { get; set; }
++            public string Name { get; set; }
++            public HealthStatus Status { get; set; }
++        }
++        public enum IconVariant {
++            Filled = 1,
++            Regular = 0,
++        }
++        public interface IModelNameParameter
++        public interface IResourceWithWaitSupport : IResource
+         public static class KnownResourceStates {
++            public static readonly IReadOnlyList<string> TerminalStates;
++            public static readonly string Waiting;
+         }
+-        public class ProjectResource : Resource, IResource, IResourceWithArgs, IResourceWithEndpoints, IResourceWithEnvironment, IResourceWithServiceDiscovery
++        public class ProjectResource : Resource, IResource, IResourceWithArgs, IResourceWithEndpoints, IResourceWithEnvironment, IResourceWithServiceDiscovery, IResourceWithWaitSupport
++        [DebuggerDisplayAttribute("Type = {GetType().Name,nq}, Type = {Type}")]
++        public sealed class ResourceCommandAnnotation : IResourceAnnotation {
++            public ResourceCommandAnnotation(string type, string displayName, Func<UpdateCommandStateContext, ResourceCommandState> updateState, Func<ExecuteCommandContext, Task<ExecuteCommandResult>> executeCommand, string? displayDescription, object? parameter, string? confirmationMessage, string? iconName, IconVariant? iconVariant, bool isHighlighted);
++            public string? ConfirmationMessage { get; }
++            public string? DisplayDescription { get; }
++            public string DisplayName { get; }
++            public Func<ExecuteCommandContext, Task<ExecuteCommandResult>> ExecuteCommand { get; }
++            public string? IconName { get; }
++            public IconVariant? IconVariant { get; }
++            public bool IsHighlighted { get; }
++            public object? Parameter { get; }
++            public string Type { get; }
```

Comment:
And similarly for the `WithCommand` method, the parameter should be `id` rather than `type`. Seems bad form to have a property/parameter named "type" in .NET that isn't actually a `Type`. My understanding is that this to uniquely identify this command vs. other commands, right?

Reply:
Yes. It is named type in the proto. We could rename it everywhere. It would be a source breaking change on the proto but not wire breaking (the proto cares about the field id, not the name on the wire)

Would the new name be Id or Name? We use name to identify many other things in aspire.

@drewnoakes

Reply:
Yeah maybe `name` is the best option here.

Reply:
PR: https://github.com/dotnet/aspire/pull/6350